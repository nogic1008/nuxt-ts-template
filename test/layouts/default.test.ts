import {
  createLocalVue,
  mount,
  shallowMount,
  RouterLinkStub,
  Wrapper
} from '@vue/test-utils'
import Buefy from 'buefy'
import DefaultLayout from '~/layouts/default.vue'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('layouts/default.vue', () => {
  let wrapper: Wrapper<Vue>

  beforeEach(() => {
    wrapper = shallowMount(DefaultLayout, {
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub,
        Nuxt: true
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(DefaultLayout, {
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub,
        Nuxt: true
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
