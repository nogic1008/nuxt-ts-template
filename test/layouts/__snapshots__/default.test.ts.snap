// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts/default.vue renders correctly 1`] = `
<div>
  <nav
    aria-label="main navigation"
    class="navbar header has-shadow is-primary"
    role="navigation"
  >
    <div
      class="navbar-brand"
    >
      <a
        class="navbar-item"
        href="/"
      >
        <img
          alt="Buefy"
          height="28"
          src="~assets/buefy.png"
        />
      </a>
       
      <div
        class="navbar-burger"
      >
        <span />
         
        <span />
         
        <span />
      </div>
    </div>
  </nav>
   
  <section
    class="main-content columns"
  >
    <aside
      class="column is-2 section"
    >
      <p
        class="menu-label is-hidden-touch"
      >
        
        General
      
      </p>
       
      <ul
        class="menu-list"
      >
        <li>
          <a>
            <span
              class="icon"
            >
              <i
                class="mdi mdi-home mdi-24px"
              />
            </span>
             Home
          
          </a>
        </li>
        <li>
          <a>
            <span
              class="icon"
            >
              <i
                class="mdi mdi-lightbulb mdi-24px"
              />
            </span>
             Inspire
          
          </a>
        </li>
        <li>
          <a>
            <span
              class="icon"
            >
              <i
                class="mdi mdi-rocket mdi-24px"
              />
            </span>
             Environment
          
          </a>
        </li>
      </ul>
    </aside>
     
    <div
      class="container column is-10"
    >
      <nuxt-stub />
    </div>
  </section>
</div>
`;
